lib/appsignal/oban.ex

```diff
@@ -112,8 +112,16 @@ defmodule Appsignal.Oban do
       |> @span.set_attribute("state", to_string(state))
 
     # The `:result` metadata key was added in Oban v2.5.0.
+    # This refers to the Oban.Worker 'result' type that controls whether a job
+    # is treated as a success or a failure.
     if Map.has_key?(metadata, :result) do
-      @span.set_attribute(span, "result", inspect(metadata[:result]))
+      {result, reason} = oban_result_and_reason(metadata[:result])
+
+      @span.set_attribute(span, "result", result)
+
+      if reason do
+        @span.set_attribute(span, "result_reason", reason)
```

Comment:
The `reason` is not guaranteed to be a string, is it? ([typing](https://hexdocs.pm/oban/Oban.Worker.html#t:result/0) says `term()`, so, it can be anything) Either we `inspect` it (like we did for `metadata[:result]` before this change) or we need to handle the case where it's not a string.

Reply:
I added a commit to improve the handling.